src/struct_mv/struct_communication.c

```diff
@@ -843,17 +843,15 @@ hypre_InitializeCommunication( hypre_CommPkg     *comm_pkg,
     * since cannot access host memory from device */
    HYPRE_Int alloc_dev_buffer = 0;
    /* In the case of running on device and cannot access host memory from device */
-#if defined(HYPRE_USING_GPU)
+   /* WM: question - what's the right thing to do for kokkos/raja here? */
 #if defined(HYPRE_USING_RAJA) || defined(HYPRE_USING_KOKKOS)
    alloc_dev_buffer = 1;
 #elif defined(HYPRE_USING_CUDA) || defined(HYPRE_USING_HIP) || defined(HYPRE_USING_SYCL)
    alloc_dev_buffer = (hypre_HandleStructExecPolicy(hypre_handle()) == HYPRE_EXEC_DEVICE);
 #elif defined(HYPRE_USING_DEVICE_OPENMP)
    alloc_dev_buffer = hypre__global_offload;
 #endif
-#endif
 
-#if defined(HYPRE_USING_GPU)
    if (alloc_dev_buffer)
    {
       send_buffers_device = hypre_TAlloc(HYPRE_Complex *, num_sends, HYPRE_MEMORY_HOST);
```

Comment:
This hasn't been touched in this PR, but I'm curious about this line. It seems we want to allocate memory for a device variable, but `HYPRE_MEMORY_HOST` is being passed. Shouldn't it be `HYPRE_MEMORY_DEVICE`? @liruipeng

Reply:
No. This is `HYPRE_Complex **`, i.e., pointer to pointers

Reply:
Oh yes! I missed that, thanks!